examples/camera-app/linux/src/camera-device.cpp

```diff
@@ -508,7 +510,7 @@ CameraError CameraDevice::StartAudioStream(uint16_t streamID)
     }
 
     int channels   = it->audioStreamParams.channelCount;
-    int sampleRate = it->audioStreamParams.sampleRate;
+    int sampleRate = static_cast<int>(it->audioStreamParams.sampleRate);
```

Comment:
I think this loses precision - this is a uint32 in the spec, right? I don't think Andrei needs to fix here since this doesn't introduce the problem, but might want to check with the original author is or whether this is a reasonable concern.

Reply:
From my side I updated what the code was doing in a way that clang also accepts it. This does seem to show a real problem. In practice I believe the sample rate is unlikely to fill a 31-bit integer anyway (it will be 48K tops generally I imagine, maybe 192K for really HiFi but I doubt it. so seems ok...ish. Not spec-ok  but probably practical-ok.

Reply:
Resolving to get PR to merge, but valid concern. @yufengwangca @pidarped